remote.go

```diff
@@ -318,7 +319,17 @@ func (r *Remote) fetch(ctx context.Context, o *FetchOptions) (sto storer.Referen
 		return nil, err
 	}
 
-	req.Wants, err = getWants(r.s, refs)
+	wantRefs, err := getWants(r.s, refs)
+	if err != nil {
+		return nil, err
+	}
+
+	var wants []plumbing.Hash
+	wants = append(wants, o.Hashes...)
+	wants = append(wants, wantRefs...)
+
+	req.Wants = wants
```

Comment:
Could we make this all one variable in memory rather than two or is there a specific reason that we have it separate? 

```
req.Wants = make([]plumbing.Hash, 0)
req.Wants = append(req.Wants, o.Hashes...)
req.Wants = = append(req.Wants, wantRefs...)
```

Reply:
Good call. I originally had it that way and second guessed before I made the commit.